src/doc/manifest.md

```diff
@@ -623,6 +623,42 @@ includes them.
 You can read more about the different crate types in the
 [Rust Reference Manual](https://doc.rust-lang.org/reference/linkage.html)
 
+# The `[augment]` Section
+
+This section of Cargo.toml can be used to [override dependencies][replace] with
+other copies. The syntax is similar to the `[dependencies]` section:
+
+```toml
+[augment.crates-io]
+foo = { git = 'https://github.com/example/foo' }
+bar = { path = 'my/local/bar' }
+```
+
+The `[augment]` table is made of dependency-like sub-tables. Each key after
+`[augment]` is a URL of the source that's being augmented, or `crates-io` if
+you're modifying the https://crates.io registry. In the example above
+`crates-io` could be replaced with a git URL such as
+`https://github.com/rust-lang-nursery/log`.
+
+Each entry in these tables is a normal dependency specification, the same as
+found in the `[dependencies]` section of the manifest. The dependencies listed
+in the `[augment]` section are resolved and used to augment the source at the
+URL specified. The above manfiest snippet augments the `crates-io` source (e.g.
+crates.io itself) with the `foo` crate and `bar` crate.
+
+Sources can be augmented with versions of crates that do not exist, and they can
+also be augmented with versions of crates that already exist. If a source is
+augmented with a crate version that already exists in the source, then the
+source's original crate is replaced.
+
+More information about overriding dependencies can be found in the [overriding
+dependencies][replace] section of the documentation and [RFC 1969] for the
+technical specification of this feature. Note that the `[augment]` feature will
+first become available in Rust 1.19, set to be released on 2017-08-27.
+
+[RFC 1969]: https://github.com/rust-lang/rfcs/pull/1969
+[replace]: specifying-dependencies.html#overriding-dependencies
+
```

Comment:
Regarding docs, we have an overwhelming number of options to *override* a package from crates.io:

* `augment`
* `replace`
* path overrides
* path dependencies 
* vendoring

Perhaps we could create a new section "using non crates io depedendencies", which gives a flow chart style guide of choosing a right tool for the problem?